runtime/parser/parser.go

```diff
@@ -179,8 +179,7 @@ func ParseTokenStream[T any](
 	result, err := parse(p)
 	if err != nil {
 		p.report(err)
-		var zero T
-		return zero, p.errors
+		return result, p.errors
```

Comment:
Was there a particular reason for changing the behaviour here and returning the result even if there is an error?